src/composables/useArrowNavigation.js

```diff
@@ -0,0 +1,153 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ref } from 'vue'
+
+/**
+ * Mount navigation according to https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus
+ * ArrowDown or ArrowUp keys to move through the itemElements list
+ * Enter key to select focused element
+ * Escape key to return focus back to the default element
+ *
+ */
+export function useArrowNavigation() {
+	const listRef = ref(null)
+	const defaultRef = ref(null)
+	const itemElements = ref([])
+	const focusedIndex = ref(null)
+
+	/**
+	 * Add event listeners for navigation list and set a default focus element
+	 *
+	 * @param {import('vue').Ref} listElementRef component ref to mount navigation
+	 * @param {import('vue').Ref} defaultElementRef component ref to return focus to
+	 */
+	function mountArrowNavigation(listElementRef, defaultElementRef) {
+		// depending on ref, listElementRef could be either a component or a DOM element
+		listRef.value = listElementRef?.$el ?? listElementRef
+		defaultRef.value = defaultElementRef
+
+		listRef.value.addEventListener('keydown', (event) => {
+			if (itemElements.value?.length) {
+				if (event.key === 'ArrowDown') {
+					focusNextElement(event)
+				} else if (event.key === 'ArrowUp') {
+					focusPrevElement(event)
+				} else if (event.key === 'Enter') {
+					focusFirstElementIfNotFocused(event)
+				} else if (event.key === 'Escape') {
+					focusDefaultElement(event)
+				}
+			}
+		})
+	}
+
+	/**
+	 * Collect all DOM elements specified by selector
+	 *
+	 * @param {string} selector selector to look for
+	 */
+	function initializeNavigation(selector) {
+		itemElements.value = listRef.value.querySelectorAll(selector)
+		focusedIndex.value = null
+	}
+
+	/**
+	 * Focus natively the DOM element specified by index
+	 *
+	 * @param {object} index the item index
+	 */
+	function nativelyFocusElement(index) {
+		focusedIndex.value = index
+		itemElements.value[index].focus()
+	}
+
+	/**
+	 * Focus the default component ('focus' method should be exposed)
+	 *
+	 * @param {Event} event Keydown event
+	 */
+	function focusDefaultElement(event) {
+		if (focusedIndex.value !== null) {
+			event.preventDefault()
+			focusedIndex.value = null
+			defaultRef.value.focus()
+		}
+	}
+
+	/**
+	 * Focus the first element if not focused yet, otherwise proceed
+	 *
+	 * @param {Event} [event] Keydown event to prevent, if defined and not focused yet
+	 *
+	 * @return {boolean} If first element was focused or not
+	 */
+	function focusFirstElementIfNotFocused(event) {
+		if (focusedIndex.value === null) {
```

Comment:
`if (focusedIndex.value !== 0)`

Because in case the user is focusing on `y`th list element , clicks back to the searchBox and hits enter again, the `focusedIndex.value === y `and nothing will happen

Reply:
Fixed a bit differently: `focusedIndex` should be reset after each `blur` event, which is not an iteration over list